pandevice/base.py

```diff
@@ -1388,35 +1443,46 @@ def _set_reference(self, reference_name, reference_type, reference_var, exclusiv
                 elif hasattr(references, "__iter__") and self in references:
                     if reference_name is not None and str(getattr(obj, reference_type.NAME)) == reference_name:
                         continue
+                    update_needed = True
                     references.remove(self)
                     if update: obj.update(reference_var)
                 elif hasattr(references, "__iter__") and str(self) in references:
                     if reference_name is not None and str(getattr(obj, reference_type.NAME)) == reference_name:
                         continue
+                    update_needed = True
                     references.remove(str(self))
                     if update: obj.update(reference_var)
                 elif references == self or references == str(self):
                     if reference_name is not None and str(getattr(obj, reference_type.NAME)) == reference_name:
                         continue
+                    update_needed = True
                     references = None
                     if update: obj.update(reference_var)
+
         # Add new reference to self in requested object
         if reference_name is not None:
-            obj = device.find_or_create(reference_name, reference_type, *args, **kwargs)
+            obj = parent.find_or_create(reference_name, reference_type, **kwargs)
```

Comment:
Is `find_or_create` still appropriate when only checking if a change is needed?

Reply:
It looks like `find_or_create()` doesn't actually do `.create()`.  It just adds the object to the object tree if it doesn't exist.  So no modifications to this are needed for `update=False`.